packages/shared/src/components/fields/CustomSwitch.tsx

```diff
@@ -79,21 +85,21 @@ export function CustomSwitch({
         ref={leftRef}
         className={checked && 'text-theme-label-tertiary'}
       >
-        {typeof LeftContent === 'string' ? (
-          LeftContent
-        ) : (
-          <LeftContent className={classNames(leftClasses)} />
-        )}
+        {typeof leftContent === 'string'
+          ? leftContent
+          : React.cloneElement(leftContent as ReactElement, {
+              className: leftClasses,
```

Comment:
Just one question, would this not override the current node's `className`?

Reply:
Yeah this will set it always to this value, but so would the original one right?
We don't pass any existing classNames, we could extend with `classNames` to take the original if you think that's needed?

Reply:
The original one though passes it on a function component, so the consuming end will be able to combine the passed className by the CustomSwitch.

Reply:
The original one though passes it on a function component, so the consuming end will be able to combine the passed className by the CustomSwitch.

Reply:
Hmm, we can maybe deal with it as we get a component that takes className?
Don't think there is one, but let me try and add classNames think that will do the trick 🙏

Reply:
I'll leave it up to you, I am cool on taking it on as we go as we don't have anything that requires it now or any upcoming component.